### ISSUE TITLE
[BUG] - ReadFile from wrong URI with multiple virtual file systems

### ISSUE THREAD

**Issue by WaynePhillipsEA** [alefragnani/vscode-bookmarks#458]
Hi, it's Wayne Phillips here, author of the twinBASIC VS code extension.   We're experiencing an incompatibility with your extension that we hope you can assist with.

**Environment/version**

- Extension version:  v13.1.0
- VSCode version:   v1.59
- OS version: Windows 10

**Steps to reproduce**

1.  Create a twinBASIC project
2.  Try to set a bookmark, nothing happens;  no error reported in the UI, and no bookmark icon was added in the gutter.

I ran your extension under the debugger, and noted that there was an unhandled error:

<img width="713" alt="tb_bookmarkError" src="https://user-images.githubusercontent.com/25738109/128908708-049f9c5b-f24f-47fe-8b4c-322ec2a66f1b.png">

This error is caused by your extension picking up the wrong URI schema.    twinBASIC uses two virtual filesystems, as seen in the screenshot below.  The first filesystem is labelled 'BUILD CONFIGURATIONS' which has a schema of 'twinbasic-builds', and a second one labelled 'PROJECT: NewProject' with a schema of 'twinbasic'.

<img width="487" alt="tb_bookmarkError2" src="https://user-images.githubusercontent.com/25738109/128909582-16b057d9-cf5d-4ce3-966d-36a785f5033b.png">

The file open in the editor here actually has a full URI of twinbasic:/NewProject/Sources/MainModule.twin, but your extension is wrongly trying to use the schema from the first virtual filesystem, twinbasic-builds, as seen in the first screenshot.

If you need any further information, please let me know.  Thanks.

**Comment by alefragnani** [comment]
Hi @WaynePhillipsEA ,

I’m sorry to say but yes, the extension does not support multiple `scheme` on the same workspace, at least not yet.

I already have plans to update that, and I guess it could work for your as well. I’ll let you know about the progress and maybe you could try out when released.

Hope this helps

**Comment by WaynePhillipsEA** [comment]
Absolutely!  I will be ready to test as soon as you have that implemented.  Thanks.

**Comment by KarlSevenSeven** [comment]
Hi, I'm a twinBASIC user.
Bookmarks are indispensable for me and twinBASIC doesn't have an own system.
So I ask if there is progress regarding the multiple scheme thing.
